backend/backend_main.py

```diff
@@ -0,0 +1,79 @@
+import os 
+from fastapi import FastAPI
+from pydantic import BaseModel
+from contextlib import asynccontextmanager
+from llama_index.core import PromptTemplate
+
+from config import load_vector_db
+from fetch_response import get_chatgpt_response
+class UserInput(BaseModel):
+    user_input: str
+
+
+VECTOR_INDEX = None
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    global VECTOR_INDEX
+    VECTOR_INDEX = load_vector_db()
+    yield
+
+    # Shut down code goes here
+    print("Shut Down")
+
+
+app = FastAPI(lifespan=lifespan)
+
+@app.get("/")
+async def read_root():
+    return {"message": "Welcome to the API. Please use the POST method to get responses."}
+
+
+@app.post("/")
+async def respond_to_user_input(user_input: UserInput):
+    # Placeholder response generation logic
+    response = generate_answer(user_input.user_input)
+    return {"response": response}
+
+
+def generate_answer(question, num_of_context=5)->str:
+    question = question.replace('"',"'")
+    retriever = VECTOR_INDEX.as_retriever(similarity_top_k=num_of_context)
+    retrieved_nodes = retriever.retrieve(question)
+
+    context = ""
+    
+    for retrieved_node in retrieved_nodes:
+        context += f"{retrieved_node.get_content()} \n\n"
+    
+    template = f"""
+    You are his holiness the 14th Dalai Lama.
+    
+    Follow these guidelines when answering the questions:
+    
+    - Answer the question based on the given contexts (some of which might be irrelevant).
+    - Be concise and precise.
+    - Answer directly, without adding any extra words.
+    - Be careful of the language, ensuring it is respectful and appropriate.
+    - If you do not have a proper answer from the context, respond with "I dont have enough data to provide an answer."
+    - Do not give a response longer than 3000 tokens.
```

Comment:
Replace this sentence " You are his holiness the 14th Dalai Lama." and add "Your task is to answer the question based on the given context.".